### DISCUSSION TITLE
[Closed] Extend ColorAnimationExtensions with 'TextColorTo' for ITextStyle

### DISCUSSION THREAD

**Discussion by PieEatingNinjas** [CommunityToolkit/Maui#203] [New Feature Discussions]
Currently the `ColorAnimationExtensions` class has 1 method `BackgroundColorTo` which helps animating the color transition of a `VisualElement`'s `BackgroundColor`.

The [ITextStyle](https://github.com/dotnet/maui/blob/45ba05e1a5c452966073d9101f21712450a61838/src/Core/src/Core/ITextStyle.cs) interface - which is implemented by things like `ILabel`, `IButton`, `IEntry`, ... - defines a `TextColor` property of type `Color`. I can imagine devs might want to animate the transition of the text color of controls that implement this interface.

That's why I want to propose adding a `TextColorTo` extension method for the `ITextStyle` interface:
```csharp
Task TextColorTo(this ITextStyle, Color color, uint rate = 16, uint duration = 250, Easing? easing = null);
```

This would allow us to use it for any UI control that implements ITextStyle, like ILabel, IButton, IEntry, etc.
```csharp
var label = new Label();
await label.TextColorTo(Colors.Red, 16, 1500, Easing.SinIn);

var button = new TextButton();
await button.TextColorTo(Colors.Green, 16, 1500);

var entry = new Entry();
await entry.TextColorTo(Colors.Pink);
```

Let's discuss! 🙂

**Comment by TheCodeTraveler** [comment]
Love it! 

Let's see if we get any more feedback from the community on this in the next few days, then open a [New Feature Proposal](https://github.com/CommunityToolkit/Maui/issues/new?assignees=&labels=new%2C+proposal&template=open-a-new-feature-proposal.md&title=%5BProposal%5D+)!

**Reply by PieEatingNinjas** [reply]
What do you think? Should I create a New Feature Proposal?

**Reply by bijington** [reply]
> What do you think? Should I create a New Feature Proposal?

I vote yes

**Reply by TheCodeTraveler** [reply]
Agreed!

**Reply by PieEatingNinjas** [reply]
Will create proposal later today or tomorrow! 🙂

**Comment by bijington** [comment]
I really like this! 

To overcomplicate the situation is there any sense in considering whether we could combine animations together?

I get that we could probably call something like the following:

```csharp
var label = new Label();
await Task.WhenAll(
    label.TextColorTo(Colors.Red, 16, 1500, Easing.SinIn),
    label.BackgroundColorTo(Colors.Orange, 16, 1000, Easing.SpringIn));
```

This may well be enough or a user could write their own for more complex scenarios. Or we further expand the compound animations coming soon. This concept is starting to get me thinking about whether this could expose a new way of doing things for the compound animation proposal too.

**Reply by TheCodeTraveler** [reply]
Oooh - yea, this sounds like a good one to look into once we've added [CompoundAnimations](https://github.com/CommunityToolkit/Maui/issues/182)!

Since you're the Proposal Champion for CompoundAnimations, I'll defer that decision to you Shaun. I'd say for now it'd be good idea to add a comment to the CompoundAnimations Proposal so that we don't forget!

**Reply by bijington** [reply]
Good call! I will add a comment there now!

**Comment by TheCodeTraveler** [comment]
Promoted to [Proposal](https://github.com/CommunityToolkit/Maui/issues/211) 🎉 

Adding `[Closed]` to title.

**Comment by CliffAgius** [comment]
Closed as moved to Proposal #211
